fvm/meter/interaction_meter.go

```diff
@@ -117,14 +132,31 @@ func GetStorageKeyValueSizeForTesting(
 	return getStorageKeyValueSize(storageKey, value)
 }
 
-func (m *InteractionMeter) GetStorageUpdateSizeMapForTesting() MeteredStorageInteractionMap {
-	return m.storageUpdateSizeMap
+func (m *InteractionMeter) GetStorageRWSizeMapForTesting() (
+	reads MeteredStorageInteractionMap,
+	writes MeteredStorageInteractionMap,
+) {
+	return m.reads, m.writes
 }
 
+// Merge merges the child interaction meter into the parent interaction meter
+// Prioritise parent reads because they happened first
+// Prioritise child writes because they happened last
 func (m *InteractionMeter) Merge(child InteractionMeter) {
-	for key, value := range child.storageUpdateSizeMap {
-		m.storageUpdateSizeMap[key] = value
+	for key, value := range child.reads {
+		_, parentRead := m.reads[key]
+		if parentRead {
+			continue
+		}
+
+		m.reads[key] = value
+		m.totalStorageBytesRead += value
+	}
+
+	for key, value := range child.writes {
+		parentWriteBytes := m.writes[key]
+		m.totalStorageBytesWritten -= parentWriteBytes
```

Comment:
⚠️  Note, this might be negative. If the write is to reduces the data size to be stored or even completely delete the registers, then totalStorageBytesWritten might become negative, will it cause underflow?

Reply:
That cannot happen. I extracted the relevant code into a method and added comments to better illustrate why. Please let me know if it clearer now.